### ISSUE TITLE
Implicit generalization disregards the implicit status of binders

### ISSUE THREAD

**Issue by kyoDralliam** [coq/coq#15398]
#### Description of the problem

The documentation of implicit generalization mentions in its BNF that a name for a binder can be provided inside brackets `` `({n} : T)`` but the specification of this construct seems to be missing in the documentation. @silene suggested [on zulip]() that this construct should make the binder a maximally-inserted implicit (so similar to `` `{ n : T }`` ?). However, in practice the argument seems to be left untouched:
```Coq
Definition f `({n} : nat) : nat := n.
About f.
(*
f : nat -> nat

f is not universe polymorphic
Arguments f _%nat_scope
f is transparent
*)
```

#### Coq Version
Tested on Coq 8.11 and 8.14

**Comment by SkySkimmer** [comment]
Introduced in ff03e8dd0de507be82e58ed5e8fd902dfd7caf4b 
back then implicit generalization looked like `[ x : type ]` (x explicit) or `[ type ]` (x implicit) or `[ {x} : type ]` (x implicit) IIUC
After cf69befd5678b6827126ef0a2b89218ea7b02c89 it changed to parse `x:type` and `{x}:type` to the same value, with implicit status controlled by surrounding `[]` or `{{}}` (implicit, eg `{{ x : type }}`) and `(())` (explicit, eg `((x : type))`, `(({x}:type))` still makes x explicit).
The modern syntax dates from c74f11d65b693207cdfa6d02f697e76093021be7 and dropped the ability to make generalized variables explicit.

In conclusion this seems like legacy syntax and should probably be dropped.
